gems/smithy-model/lib/smithy-model/shapes.rb

```diff
@@ -198,6 +211,7 @@ class StructureShape < Shape
         def initialize(options = {})
           super
           @members = {}
+          @members_by_name = {}
```

Comment:
Basically - any shapes that involves members - we want to retain their member names to reference for (de)serialization process. (AKA the location_name in Shape ref)

Reply:
Yea. I think we should choose a different name though..